TaskManager.go

```diff
@@ -0,0 +1,106 @@
+package interview
+
+import (
+	"container/list"
+	"database/sql"
+	"sync"
+	"time"
+)
+
+type TaskManager struct {
+	queue    list.List // container/list
+	running  map[Task]struct{}
+	blocked  map[Task]struct{}
+	conn     *sql.Conn
+	Incoming chan Task // alternative way to schedule task
+}
+
+var instance *TaskManager = nil
+
+func GetInstance() *TaskManager {
+	new(sync.Once).Do(func() {
+		instance = new(TaskManager)
+		instance.queue = list.List{}
+		instance.running = map[Task]struct{}{}
+		instance.blocked = map[Task]struct{}{}
+		instance.Incoming = make(chan Task)
+		for task := range instance.Incoming {
+			instance.SkeduleTask(task)
+		}
+		go func() {
+			for {
+				instance.checkRun()
+			}
+		}()
+	})
+	return instance
+}
+
+func (me TaskManager) RunTask(task Task) {
```

Comment:
Generally, if you need to modify me. or any of it value attributes - operate it via pointer.